### ISSUE TITLE
[Bug]: purple-googlechat is unable to login

### ISSUE THREAD

**Issue by visious25** [EionRobb/purple-googlechat#75]
Hi, purple-googlechat is unable to login. Please check this [link](https://github.com/Intika-Pidgin/Pidgin/issues/3), because I hate to repeat myself.

**Comment by bminer** [comment]
The OAuth link in the [YouTube video description](https://www.youtube.com/watch?v=hlDhp-eNLMU) no longer works.
![image](https://user-images.githubusercontent.com/73755/226733444-7d764b5a-838f-4c0a-a5d6-20a07b5173d5.png)

**Comment by bminer** [comment]
Side note, it would be super cool if you could use the Gnome Online Accounts to pull in any Google account credentials automatically!

**Comment by fulalas** [comment]
> The OAuth link in the [YouTube video description](https://www.youtube.com/watch?v=hlDhp-eNLMU) no longer works. 

I'm facing the same issue. :(

**Comment by grimmy** [comment]
Yeah unfortunately google disabled the authentication mechanism that this plugin was using. Also this plugin can't use a normal google login method because then it wouldn't have access to the google chat api as google doesn't export it to third party oauth applications.

There's been a considerable amount of work put in in the background, but there's no solution yet. However it looks like there may be a solution but this plugin is going to need to need a significant amount work to get to that point.

TL;DR stuff is broke, it's going to take some time to get it all sorted.

**Comment by bertiebaggio** [comment]
Is there anything end users can do to help facilitate sorting out potential solutions? :)

**Comment by grimmy** [comment]
Not really it's a mess. But if you can think of any other google chat clients that aren't the ios version, android version, or the web version that _might_ be useful.

**Comment by bertiebaggio** [comment]
I was getting all keyed up to mention [mautrix/googlechat](https://github.com/mautrix/googlechat), a GChat puppeting bridge for Matrix, but glanced at [auth.py](https://github.com/mautrix/googlechat/blob/e2eb528745466468f059c506c22e500e0cd832aa/maugclib/auth.py) and saw:

> Logging into Hangouts using OAuth2 requires a private scope only whitelisted
> for certain clients. This module uses the client ID and secret from iOS, so it
> will appear to Google to be an iOS device

and

https://github.com/mautrix/googlechat/issues/85

Alas. 

It seems that [tulir](https://github.com/tulir) is looking at it too, at least, so hopefully once a solution is found it can be implemented by all bridges. If it becomes possible and helpful for end-users to assist with traffic sniffing, testing, etc I'm sure you'll have some volunteers!

**Comment by grimmy** [comment]
> I was getting all keyed up to mention [mautrix/googlechat](https://github.com/mautrix/googlechat), a GChat puppeting bridge for Matrix, but glanced at [auth.py](https://github.com/mautrix/googlechat/blob/e2eb528745466468f059c506c22e500e0cd832aa/maugclib/auth.py) and saw:

If you look closely at that repository you'll see my name too :)

**Comment by MrLint** [comment]
I really hope there is some kinda work around. I wanted to get my son setup on pidgin without having to deal with the browser all the time

**Comment by MrLint** [comment]
So I know the plugin is looking for the oauth, is there a way to connect with the app password setup that google still has?

**Comment by Holly275** [comment]
Any update on a fix to get Google Chat to work?  I too hate the browser window.  I miss my Pidgin and now have to keep Pidgin and Chat open at the same time.  Please, please, please :)

**Comment by Launcelot555** [comment]
Looks like they resolved it by using the web app API? https://github.com/mautrix/googlechat/pull/89

**Comment by grimmy** [comment]
> Looks like they resolved it by using the web app API? [mautrix/googlechat#89](https://github.com/mautrix/googlechat/pull/89)

Kind of. If you look closely you'll see I'm the one that did the work. The TL;DR is that the web login is used instead of the oauth out of band auth that was previously used. That requires an embedded browser, something we don't have and can't easily make work with pidgin right now. That said, I suppose we could hack together a patch that would allow you to enter the 6 cookies that are necessary and with a few other changes it could work, but that's less than ideal.

**Comment by bertiebaggio** [comment]
Do the cookies have a reasonable expiry time?

Cookies are at least plain text, so it's still kind of just a string like an oauth token if you squint your eyes closed hard enough. Still a pain to grab/paste nicely compared to oauth, but beggars can't be choosers.

Sure would be nice if Google hadn't killed XMPP! (and, you know, everything else they've killed)

**Comment by grimmy** [comment]
Yeah they're pretty clear but you need 6 of them and they'll get refreshed automatically from what I've seen since I started working on this back in April.

The real question here is if @EionRobb would want a patch to do this. But the cookie capture procedure differs a bit between firefox and chrome and I haven't tested other browsers. See https://github.com/mautrix/googlechat/blob/master/maugclib/README.md

**Comment by bertiebaggio** [comment]
Thanks for the info! There's presumably an automated way to get the cookie info out of Firefox / Chrome like an addon/extension†? But that's getting into the realm of hacks supporting hacks...

† Though maybe not since auth info lives there and browsers care about security? I last properly looked at getting cookies out of the browser around the IE6 days it feels like, so maybe I should stop guessing

**Comment by grimmy** [comment]
> Thanks for the info! There's presumably an automated way to get the cookie info out of Firefox / Chrome like an addon/extension†? But that's getting into the realm of hacks supporting hacks...

Nope, all of the cookies are `HttpOnly` which means they can't be read from JavaScript. This is why an embedded browser is necessary.

**Comment by bertiebaggio** [comment]
I mean from a user perspective to get them in to mau/purple in a slightly easier way than devtools- I seem to be able to export them using cookies.txt. A few minutes of scripting would have the python dict from that!

**Comment by LurkerHub** [comment]
I just opened a private window, logged in, and checked the cookies on "chat.google.com"
The only ones I see are: `/OSID`, `/__Secure-OSID`, `/OTZ`, `/COMPASS`, and `/u/0/webchannel/COMPASS`.
`SSID`, `SID`, and `HSID` reside in ".google.com"

I've been authenticated to google chat in Pidgin for months, without it kicking me out.

There are several add-ons for both FF and Chrome that support exporting cookies to a text file.  I think that adding an "import cookies from file" option to the plug-in should be doable.  The more technical users can edit blist.xml directly.

**Comment by smith153** [comment]
Ugh, just realized why I haven't heard from any of my friends on pidgin... looks like was stuck waiting for auth. I guess my oath key was no longer valid (set it up a year ago). So there is still no way to purple-googlechat to work now?

**Comment by Shdwdrgn** [comment]
While my home computer lost its connection a couple months ago, I still have access to both a Windows and Linux computer which are still able to connect to google chat.  Is there any information I could gather from either of these machines which might help in solving this problem, or is it simply a matter of time before those machines also expire?

In the meantime, I finally gave up and just set up my own XMPP server.  Obviously can't trust Google or any of the other sites to play nicely.

**Comment by Jhojoyms** [comment]
@EionRobb please master, help us for this problem

**Comment by Hawk777** [comment]
Is it possible to do the authentication process in a browser and copy the cookies over using the plugin as it exists today? Or does that workaround also require a patch to the plugin to make it use those cookies?

**Comment by john-d-murphy** [comment]
Understood that this is still an ongoing issue, just putting my hand up and saying that i attempted this today and google is still being annoying.

**Comment by dchmelik** [comment]
I have same problem.  Embedded browser surely can't be an option because purple-googlechat also was for BitlBee, which is text-only (unless you have some text-only web browser you can embed that's going to get the information).

**Comment by EionRobb** [comment]
I have a maybe solution - but it's a maybe and needs a bit of verification from someone else to see if it's viable.

1. Open [this link](https://accounts.google.com/v3/signin/identifier?client_id=936475272427.apps.googleusercontent.com&delegated_client_id=576267593750-sbi1m7khesgfh1e0f2nv5vqlfa4qr72m.apps.googleusercontent.com&device_name=Pidgin+Goole+Chat+Plugin&hl=en-NZ&kdlc=1&kss=uca&lib_ver=3.4&redirect_uri=com.google.sso.576267593750-sbi1m7khesgfh1e0f2nv5vqlfa4qr72m%3A%2FauthCallback%3Flogin%3Dcode&sarp=1&scope=https%3A%2F%2Fwww.google.com%2Faccounts%2FOAuthLogin+https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fuserinfo.email&sfas=1&system_version=15.7.5&flowName=GlifDesktopChromeSync&continue=https%3A%2F%2Faccounts.google.com%2Fprogrammatic_auth%3Fdelegated_client_id%3D576267593750-sbi1m7khesgfh1e0f2nv5vqlfa4qr72m.apps.googleusercontent.com%26hl%3Den-GB&theme=glif&) in an incognito browser window
2. Open dev tools, switch to Network tab and enable 'Preserve log'
3. Login until you get the "401 error" page
4. Look for the last "batchexecute" in the network list
5. Look for an X-Chrome-Id-Consistency-Response header which should have "authorization_code=4/...."
6. Copy the code, starting with and including the   4/
7. Cross fingers
8. Paste the 4/... code into Pidgin in the popup

**Comment by bssb** [comment]
I tried this method, and was able to copy the 4/ auth code, but login fails with "Bad Request".

**Comment by pavolgaj** [comment]
At first, 'Preserve log' should be enabled in dev tools - otherwise, it'll be cleaned after the page reload. And it works also in non-incognito.
But, I got "Bad Request" when I wanted to log in to Pidgin as @bssb already mentioned.

**Comment by Hawk777** [comment]
I tried this, in both Firefox and Chrome, in both incognito/private and not, and for me it failed in a different way: the login flow ended with a 400 not a 401, and there were no `X-Chrome-Id-Consistency-Response` headers in any of the transactions.

**Comment by mkresch** [comment]
Sorry to pile on; but...   I also tried.  In chrome only.   Non-incognito, I got the 4/ auth code and had a failed long with "Bad Request" as per @bssb .   In incognito mode I got a 400 error with no X-Chrome-Id-Consistency-Response as per @Hawk777 .

That said, thank you for still trying to get this working... Much appreciated!

**Comment by xwin** [comment]
@EionRobb , I was wondering if the login token can be obtained using this URL https://developers.google.com/oauthplayground/. I tried and could get a token similar to what I saw before. But I am not sure for which endpoint to request it.

**Comment by huh02** [comment]
Logging into Pidgin stopped working for me about a week ago. I assume because the ID/OAuth has expired. The only option for me at the moment that  works on Win7 is a Google Chat PWA + program that creates a button to minimize to the system tray. Another option was Android X86 in Virtualbox, but it was too cumbersome for autorun, window size, etc.

**Comment by Holly275** [comment]
Sorry to be a pest. Any update on the 404 error so I can again use Pidgin and Chat?  I really miss it and I keep seeing updates but doubt that is has the fix?

**Comment by xwin** [comment]
I believe the issue is related to this change - https://developers.google.com/identity/protocols/oauth2/resources/oob-migration
I created a secret file based on the values in the source code and it results is saying that the client in non-secure and to upgrade to secure client.
My old token still works but I don't think new token can be created using old methods.
IMHO the simplest way would be to register the app and use python libraries referenced on this page:
https://developers.google.com/workspace/chat/authenticate-authorize-chat-user#desktop-app

**Comment by EionRobb** [comment]
@xwin sadly, registering an app doesn't allow us access to all of the APIs we need. 

Unless someone following this thread is/knows someone at Google who can grant special permissions to our app to be able to do so? 😄

**Comment by xwin** [comment]
Yes, I have tried to call a chat api from python but failed. I was able to successfully authenticate but calling the api is not allowed. I do have access to the workspace account but no client_secret file so I can't authenticate against it.
Looks like this project is dead until Google allows mere mortals to use their chat API.

**Comment by dchmelik** [comment]
I didn't receive that response header.

**Comment by tukanos** [comment]
> @xwin sadly, registering an app doesn't allow us access to all of the APIs we need.
> 
> Unless someone following this thread is/knows someone at Google who can grant special permissions to our app to be able to do so? 😄

What Google plan are you using?  Is that for a free account or paid one?

**Comment by tukanos** [comment]
It also looks like this[ electron application - google-chat-linux](https://github.com/squalou/google-chat-linux) does work.  I tried to cruise through the code but could not find where they have the code for oauth, but I'm no javascript/electron guy.

**Comment by bmomjian** [comment]
I think Electron works because they have a web browser that can do oAuth, which Pidgin does not have.  However, at the risk of proposing a crazy idea, I looked into using the Mutt email client to get email via IMAP from Google.  The solution was to create a client id and secret in Google and store those in Mutt.  Mutt doesn't have a web browser either; see  https://hackmd.io/@linD026/mutt-oauth2   It would be interesting to know how Mutt does it with no web browser.

**Comment by tukanos** [comment]
> I think Electron works because they have a web browser that can do oAuth, which Pidgin does not have. However, at the risk of proposing a crazy idea, I looked into using the Mutt email client to get email via IMAP from Google. The solution was to create a client id and secret in Google and store those in Mutt. Mutt doesn't have a web browser either; see https://hackmd.io/@linD026/mutt-oauth2 It would be interesting to know how Mutt does it with no web browser.

Great idea, I have checked [Evolution's google OAuth2](https://gitlab.gnome.org/GNOME/evolution/-/commit/24c65e30261ef3aae2a6583e9697574d2f35b3e4) and it also looks good.

**Comment by Holly275** [comment]
Just checking, non technical user, I assume we still can't use pidgin for Google Chat? Hoping I missed something. I still miss my pidgin.

**Comment by bmomjian** [comment]
@Holly275 Yes, that is my assumption.  I have been using Google Chat in my browser.

**Comment by tramtrist** [comment]
Really miss being able to use this :(
Ive tried cracking the OAuth issue over the last couple month through various methods and have failed each time :/

**Comment by tukanos** [comment]
> what a crap in 2024 that's the reason i still use irc - fuck these gogles and alikes with their stupid "developers" - i knew they were insane, this is just another proof..

Yes, irc is great, but the issue is that you can not use it for everything.  I have decided to take other route.  I'm giving the chat constantly the worst possible review (company is paying google) so the motivation should be $$$.

**Comment by tukanos** [comment]
> ha, quite an idea! still, i see 4,6 stars for that dysfunctional kreppe on appstore. i remember friend sending me a door code and then, once by the door, i couldn't see it without mobile data on, because that shit-program showed "can't load chat" or something although it had it just half a hour ago. what a dork wrote that software...:o

It is always good to see where the motivation lies.  The motivation of Google and other giga-corporations is to have you always on internet, monitor you, have your focus and thus earning $$$.   If the users lower their ratings to a garbage level then less people will be using them and searching for alternatives which will force the corporations to take action(s) in the long run.

**Comment by Shdwdrgn** [comment]
I don't know if this will be helpful, but I seem to have a unique situation here that maybe we can exploit?  I work on someone's computer who strangely is still able to sign in the google chat through pidgin.  I originally set this up for her several years ago so we could chat, and I have since moved over to my own instance of OpenFire to provide an XMPP chat for us to use, however when she rebooted this morning I noticed that her connection to google chat still appears to be logging in.  I haven't taken any steps to verify this yet but I know her login was working long after mine had failed and I had to switch to their web interface (before I found another client), so for all appearances I do believe it is somehow still alive.

The point of all this is that I do have access to remote connect into her machine when she's not using it, and I could do some poking around if there is any particular information I could collect that might be useful here?  Let me know... even though I don't use pidgin any more I still follow this bug report.

**Comment by saleslogistix** [comment]
> I don't know if this will be helpful, but I seem to have a unique situation here that maybe we can exploit? I work on someone's computer who strangely is still able to sign in the google chat through pidgin. I originally set this up for her several years ago so we could chat, and I have since moved over to my own instance of OpenFire to provide an XMPP chat for us to use, however when she rebooted this morning I noticed that her connection to google chat still appears to be logging in. I haven't taken any steps to verify this yet but I know her login was working long after mine had failed and I had to switch to their web interface (before I found another client), so for all appearances I do believe it is somehow still alive.
> 
> The point of all this is that I do have access to remote connect into her machine when she's not using it, and I could do some poking around if there is any particular information I could collect that might be useful here? Let me know... even though I don't use pidgin any more I still follow this bug report.
> 
> Edit -- just confirmed when I signed in to google's web interface... she IS still connected through pidgin, and able to send and receive messages. Wow I don't get it because last time I set her up was when we had to start copying the OAuth keys from the web browser and I haven't changed anything since then.

Yeah, old Oauth keys will still work, but only as long as you don't change the password on your Talk/Chat account.

Don't know why Google lets legacy connections in...

**Comment by Shdwdrgn** [comment]
Sorry to disagree, but my own account stopped working around the same time as most of the other folks in this bug report.  I know for a fact I did not change my password or make any other changes to my account before it stopped working in Pidgin -- I simply was unable to log in one day.  That's why I thought it was so strange that this other person's account is still working.

**Comment by antdude** [comment]
Mine still works in my old 64-bit W10 Pro. PC. I think it is because I 
never logged out and changed my old password. Knock on wood! I'll be sad 
if I can't relog in again via Pidgin. :(

**Comment by sandebert** [comment]
FWIW, I'm also still able to use this fine in Pidgin, probably because I copied my settings directory when I got a new computer. I'm on Debian Stable.

**Comment by Shdwdrgn** [comment]
Running an old Debian stable here (Buster) so there haven't been any real updates in quite some time.  When I got kicked out and found this bug report, I just assumed *everyone* was getting kicked off of pidgin at the same time.  Now I'm even more confused why some people are getting kicked out and others are still able to use it a couple years later.  Guess I'm just one of the unlucky ones.

**Comment by sfjes** [comment]
Mine continued working long after this bug was created, using my existing token, until one day I accidentally cleared all the logged in sessions on my Google account from the security dashboard. Oops. Perhaps something similar happened to you @Shdwdrgn?

**Comment by MrLint** [comment]
My connection using the old oath is also still alive.

**Comment by Holly275** [comment]
I'm still hoping I'll be able to use Pidgin again one day.  (I hate the chat web version).  My husband's computer still works but mine stopped along time ago.  (I think I had logged out/in or purchased a new computer and then couldn't get back in. Honestly don't remember.)  Just hoping someday 🙂

**Comment by Shdwdrgn** [comment]
> Mine continued working long after this bug was created, using my existing token, until one day I accidentally cleared all the logged in sessions on my Google account from the security dashboard. Oops. Perhaps something similar happened to you [@Shdwdrgn](https://github.com/Shdwdrgn)?

I'm not sure how I could have?  I never even logged in to the web chat until pidgin stopped working, and I don't do anything with other google tools.  The only other time I remember poking at the settings was back when we had to add an exception for the pidgin login, but again that only happened when pidgin stopped working for me.

**Comment by jankratochvil** [comment]
> I'm still hoping I'll be able to use Pidgin again one day. (I hate the chat web version).

You can use your own XMPP server (using [Prosody](https://prosody.im/)). It works for me from Pidgin and [Conversations](https://conversations.im/) but in the end I could not use it as I could not get authentication working for my wife's iOS.

**Comment by Hawk777** [comment]
@jankratochvil are you saying Prosody can connect to Google Chat? I took a glance at the Prosody documentation but didn’t see anything about connecting to Google Chat; if it can, I’d appreciate a pointer.

Or are you just suggesting trying to convince all one’s conversation partners to also switch?

**Comment by jankratochvil** [comment]
> [@jankratochvil](https://github.com/jankratochvil) are you saying Prosody can connect to Google Chat?

No.

> Or are you just suggesting trying to convince all one’s conversation partners to also switch?

Yes.

**Comment by Plaque-fcc** [comment]
> Or are you just suggesting trying to convince all one’s conversation partners to also switch?

You can try. Or you can switch and ask others to switch too.
Or you can keep using Google Chat, if it works for you.

For me, this plugin works. But I don't rely on it.

**Comment by BoFiS** [comment]
So wait, is this STILL an issue with new setups?  That's a real bummer...mine is still working on both my work and home machines ::knock on wood::

**Comment by EionRobb** [comment]
Still no solution yet, but have been looking at https://docs.mau.fi/bridges/python/googlechat/authentication.html and how they do it to bring through to this plugin, so all hope is not lost

**Comment by tramtrist** [comment]
Hope springs eternal

**Comment by EionRobb** [comment]
Been playing around and have things working for a fresh account in Pidgin using the cookies from the browser method.

It's not yet a great experience, so just in a separate PR for now at https://github.com/EionRobb/purple-googlechat/pull/87

**Comment by EionRobb** [comment]
> Will this also work for us lowly bitlbee people? 🤞

Yes, even people crazy enough to not endure the eye-torture of Pidgin should be able to :wink:

```
account <account id> set COMPASS_token <value>
account <account id> set SSID_token <value>
account <account id> set SID_token <value>
account <account id> set OSID_token <value>
account <account id> set HSID_token <value>
```

**Comment by tramtrist** [comment]
!!!!!!!
It also helps if you pull the correct PR 😆 
THANK YOU EIONROBB!!

**Comment by Dehela** [comment]
Giving it a try on separate account, so far so good! Really hope this update fixes this issue for good. When I accidentally lost (luckily I managed to restore it) the token few months ago, I was so depressed of the thought that I would have to switch browser, I am too used to a "proper" application at this point.

**Comment by vkareh** [comment]
I am able to login now using #87 🎉
However, contacts don't show up. Once I receive a message I'm then able to alias them (their name defaults to a number/id) and send messages to them from that point on.
This is a huge improvement in quality of life, thanks @EionRobb!

**Comment by kriukov** [comment]
Thanks, I was able to log in. Great! Will see how it works.

**Comment by EionRobb** [comment]
> With spectrum2 I couldn't find any way to supply extra user account strings for COMPASS, SSID, SID, OSID and HSID (am I just missing the way?), so I made a very messy patch to parse those from the password field. I'm not actually prompted for a password, so I stuck it directly into the sqlite field where I used to stick the OAuth token. It'd be cool if purple-googlechat could prompt for this value (a combined cookies string) and store is like it used to for the OAuth token, but I'm just happy to have connectivity working at all.

Ah, spectrum2 users might have a hard time. Will try think of a way to help them out.  At the moment, spectrum2 is hard-coded to look for specific messages from protocol plugins, to provide hard-coded settings eg https://github.com/SpectrumIM/spectrum2/blob/f547a2158bc90302ddfe09915a557d2e4ac4dfe8/backends/libpurple/main.cpp#L1640-L1651 (as opposed to bitlbee which just has a generic fake `purple0` user that you can DM with responses to popups and such)

Using the password field is clever.  I've attempted an even messier patch to try parse the cookies out from there.  If you're able to give it a whirl?  Hopefully it doesn't break what you've done too much, @sfjes

**Comment by tramtrist** [comment]
> I haven't tried this yet, but just curious: if any or all of us wanted to buy you a coffee or a car (some people have deeper pockets than I do), what's the best way to send you money? I found your sponsorship page, but it's set to a $10,000 minimum and, frankly, I'm more likely to send a coffee than a car.

Yes.  Do tell.

**Comment by EionRobb** [comment]
> I found your sponsorship page, but it's set to a $10,000 minimum and, frankly, I'm more likely to send a coffee than a car.

Whoops! 😅 

There's a couple of other sponsorship links on the plugin page, eg to buymeacoffee. I'll update the GitHub sponsor page shortly.

**Comment by tukanos** [comment]
@EionRobb Apparently I was wrong that the plugin works.  It only appears to work.  I have compiled and installed the new master version.  When I start piding in debugging mode I get an error.

When I  the chat.google.com:
```
(11:19:55) http: Request 0x56554e06d820 performed without success.
(11:19:55) googlechat: Error from server: (Unable to connect to chat.google.com: Cancelled) 
(11:19:55) http: Request 0x56554e06d720 performed without success.
(11:19:55) googlechat: Error from server: (Unable to connect to chat.google.com: Cancelled) 
(11:19:55) http: Request 0x56554e74a0e0 performed without success.
(11:19:55) http: Cannot perform another HTTP request while cancelling all related with this PurpleConnection
```

I have two compass lines even thou I'm using Chromium.  I have used the one with `/` path and it did not work.  Here is my screenshot:


![Image](https://github.com/user-attachments/assets/a819dd29-f261-4dc5-877b-e6b9e1262e48)

Do you have any ideas what could I try next?

**Comment by tukanos** [comment]
> [@EionRobb](https://github.com/EionRobb) Apparently I was wrong that the plugin works. It only appears to work. I have compiled and installed the new master version. When I start piding in debugging mode I get an error.
> 
> When I the chat.google.com:
> 
> ```
> (11:36:08) certificate: Successfully verified certificate for chat.google.com
> (11:36:09) http: Performing new request 0x5644c0a72ce0 to chat.google.com.
> 
> (11:36:09) http: Performing new request 0x5644c085fb40 to chat.google.com.
> (11:36:09) http: Performing new request 0x5644c085fc40 to chat.google.com.
> (11:36:09) account: Disconnecting account txxx@xxxxx.biz (0x5644c039d740)
> 09) http: Request 0x5644c085fc40 performed without success.
> (11:36:09) http: Request 0x56554e06d820 performed without success.
> (11:36:09) googlechat: Error from server: (Unable to connect to chat.google.com: Cancelled) 
> (11:36:09) http: Request 0x5644c085fb40 performed without success.
> (11:36:09) googlechat: Error from server: (Unable to connect to chat.google.com: Cancelled) 
> (11:36:09) http: Request 0x5644c0a72ce0 performed without success
> ```
> 
> I have two compass lines even thou I'm using Chromium. I have used the one with `/` path and it did not work. Here is my screenshot:
> 
> ![Image](https://github.com/user-attachments/assets/a819dd29-f261-4dc5-877b-e6b9e1262e48)
> 
> Do you, by any chance, have any ideas what could I try next?

Ok, I have found a workaround.  Use Firefox.  When I did that with firefox it just worked! Thank you for our work!

**Comment by Plaque-fcc** [comment]
> [@rmcmilli](https://github.com/rmcmilli) I think it should be enough to log into `chat.google.com` (using this exact address).

I just tried, and Pidgin crashed several times in the process of handling some first minutes, but eventually it started working.
